lib/internal/test_runner/yaml_to_js.js

```diff
@@ -44,9 +44,11 @@ function reConstructError(parsedYaml) {
     cause = new Error(parsedYaml.error);
     cause.code = parsedYaml.code;
   }
-  cause.stack = stack;
+  const name = parsedYaml.name ?? 'Error';
+  cause.stack = `${name}: ${parsedYaml.error}\n${stack}`;
 
   if (isTestFailure) {
+    if (!isAssertionError) delete cause.code;
     error = new ERR_TEST_FAILURE(cause, parsedYaml.failureType);
     error.stack = stack;
```

Comment:
Independent from this PR: the cause already contains the same stack. We could maybe limit it to either or?